backend/src/main/java/com/huellapositiva/application/controller/ProposalApiController.java

```diff
@@ -0,0 +1,153 @@
+import com.huellapositiva.application.exception.InvalidJwtTokenException;
+import com.huellapositiva.application.exception.ProposalNotPublished;
+import com.huellapositiva.application.exception.UserNotConfirmed;
+import com.huellapositiva.domain.actions.FetchProposalAction;
+import com.huellapositiva.domain.actions.JoinProposalAction;
+import com.huellapositiva.domain.actions.RegisterProposalAction;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.annotation.security.RolesAllowed;
+import javax.persistence.EntityNotFoundException;
+import javax.servlet.http.HttpServletRequest;
+
+@RestController
+@AllArgsConstructor
+@Tag(name = "Proposal Service", description = "The proposals API")
+@RequestMapping("/api/v1/proposals")
+public class ProposalApiController {
+
+    private final RegisterProposalAction registerProposalAction;
+
+    private final FetchProposalAction fetchProposalAction;
+
+    private final JoinProposalAction joinProposalAction;
+
+    @Operation(
+            summary = "Register a new proposal",
+            description = "Register a new proposal and link it to the logged employee",
+            tags = "proposals"
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "Ok, proposal register successful"
+                    ),
+                    @ApiResponse(
+                            responseCode = "422",
+                            description = "Unprocessable entity, in order to create a proposal the employee has to be confirmed.",
+                            content = @Content()
+                    ),
+                    @ApiResponse(
+                            responseCode = "409",
+                            description = "Conflict, could not register proposal.",
+                            content = @Content()
+                    )
+            }
+    )
+    @PostMapping("/register")
+    @RolesAllowed({"ORGANIZATION_EMPLOYEE","ORGANIZATION_EMPLOYEE_NOT_CONFIRMED"})
+    @ResponseBody
+    public void createProposal(@RequestBody ProposalRequestDto dto, HttpServletRequest req) {
+        try {
+            registerProposalAction.execute(dto, req);
+        } catch (UserNotConfirmed e){
+            throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "Employee is not confirmed yet");
+        }catch (Exception ex) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "Could not register the proposal");
```

Comment:
If we are not able to determine the error beforehand this should probably be 500 for unknown errors, otherwise catch the errors you expect